File: skybay888_ui/src/services/TicketPriceService.js
import http from "../http-common"; 

class TicketPriceService {
  getAllTicketPrices(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/ticketPrice/ticketPrices`, searchDTO);
  }

  get(ticketPriceId) {
    return this.getRequest(`/ticketPrice/${ticketPriceId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/ticketPrice?field=${matchData}`, null);
  }

  addTicketPrice(data) {
    return http.post("/ticketPrice/addTicketPrice", data);
  }

  update(data) {
  	return http.post("/ticketPrice/updateTicketPrice", data);
  }
  
  uploadImage(data,ticketPriceId) {
  	return http.postForm("/ticketPrice/uploadImage/"+ticketPriceId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new TicketPriceService();

File: src/main/java/com/skybay888/service/TicketPriceService.java
package com.skybay888.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skybay888.domain.TicketPrice;
import com.skybay888.dto.TicketPriceDTO;
import com.skybay888.dto.TicketPriceSearchDTO;
import com.skybay888.dto.TicketPricePageDTO;
import com.skybay888.dto.TicketPriceConvertCriteriaDTO;
import com.skybay888.service.GenericService;
import com.skybay888.dto.common.RequestDTO;
import com.skybay888.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface TicketPriceService extends GenericService<TicketPrice, Integer> {

	List<TicketPrice> findAll();

	ResultDTO addTicketPrice(TicketPriceDTO ticketPriceDTO, RequestDTO requestDTO);

	ResultDTO updateTicketPrice(TicketPriceDTO ticketPriceDTO, RequestDTO requestDTO);

    Page<TicketPrice> getAllTicketPrices(Pageable pageable);

    Page<TicketPrice> getAllTicketPrices(Specification<TicketPrice> spec, Pageable pageable);

	ResponseEntity<TicketPricePageDTO> getTicketPrices(TicketPriceSearchDTO ticketPriceSearchDTO);
	
	List<TicketPriceDTO> convertTicketPricesToTicketPriceDTOs(List<TicketPrice> ticketPrices, TicketPriceConvertCriteriaDTO convertCriteria);

	TicketPriceDTO getTicketPriceDTOById(Integer ticketPriceId);







}





